### PR TITLE
Change default behavior of Homepage Posts and Carousel blocks to include posts in subcategories 

### LINKED ISSUE #207: Article Block: Get articles from the sub-categories as well?
Original issue by @WPArena here: https://github.com/Automattic/newspack-theme/issues/562:

> Currently, the home page news block is not fetching articles from sub categories. It's just fetching articles from the selected category. I want that it fetch articles from parent and subcategories.

> Can you please fix it?

I think we may want to confirm if this is intuitive to the cohorts as well, and make sure it doesn't cause any unexpected behaviour on live sites.

### PR DESCRIPTION
**Pull request by DennisNDean**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
Change the default behavior of the Homepage Posts and Carousel blocks to include posts in subcategories of the selected categories. Added a toggle under the category picker to switch this behavior off. 

Generally speaking, this is more in line with how Wordpress treats parent/child categories, where a query for the parent category is inclusive of posts in the child categories, and seems more intuitive. 

Closes #207

### How to test the changes in this Pull Request:

1. Create a category with one or more child categories (i.e. Food > Tacos)
2. Categorize some posts into the parent category (i.e. Food)
3. Categorize some posts into the child category, but do not explicitly categorize them into the parent category (i.e. Tacos, but not Food)
4. Create a new page and add a Homepage Posts block
5. Under Display Settings, filter by the parent category (i.e. Food)
6. Observe that posts from the child category are also included 
7. Toggle "Include subcategories" off, observe that now only posts from the parent category are displayed
8. Do the same with the Carousel block

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
Thank you for your contribution, @DennisNDean!

It's working as described and the code looks good. A few minor changes are required before we are able to merge. They are commented inline, please let me know if you have any questions.

File: includes/class-newspack-blocks.php [hunk 1]
```diff
@@ -628,16 +628,17 @@ function ( $acc, $block ) use ( $block_name ) {
...
+		$categories             = isset( $attributes['categories'] ) ? $attributes['categories'] : array();
+		$include_subcategories  = isset( $attributes['includeSubcategories'] ) ? intval( $attributes['includeSubcategories'] ) : false;
+		$tags                   = isset( $attributes['tags'] ) ? $attributes['tags'] : array();
+		$custom_taxonomies      = isset( $attributes['customTaxonomies'] ) ? $attributes['customTaxonomies'] : array();
+		$tag_exclusions         = isset( $attributes['tagExclusions'] ) ? $attributes['tagExclusions'] : array();
+		$category_exclusions    = isset( $attributes['categoryExclusions'] ) ? $attributes['categoryExclusions'] : array();
+		$specific_posts         = isset( $attributes['specificPosts'] ) ? $attributes['specificPosts'] : array();
+		$posts_to_show          = intval( $attributes['postsToShow'] );
+		$specific_mode          = isset( $attributes['specificMode'] ) ? intval( $attributes['specificMode'] ) : false;
+		$args                   = array(
```
Comment (miguelpeixe) [inline includes/class-newspack-blocks.php:641]: Looks like there's a double-space on `$include_subcategories =`. Can you trim the extra space off from these lines?

File: includes/class-newspack-blocks.php [hunk 2]
```diff
@@ -666,6 +667,14 @@ function ( $acc, $block ) use ( $block_name ) {
 				);
 			}
 			if ( $categories && count( $categories ) ) {
+				if ( $include_subcategories === 1 ){
+					foreach  ( $categories as $parent ) {
+						$children[] = get_categories(array('child_of'=> $parent));
+						foreach ( $children[0] as $child ) {
+						$categories[] = $child->term_id;
```
Comment (miguelpeixe) [inline includes/class-newspack-blocks.php:674]: Missing a tab here

**Comment by DennisNDean** [comment]
Awesome, thank you — those changes have been made.

File: includes/class-newspack-blocks.php [hunk 3]
```diff
@@ -666,6 +667,14 @@ function ( $acc, $block ) use ( $block_name ) {
 				);
 			}
 			if ( $categories && count( $categories ) ) {
+				if ( $include_subcategories === 1 ){
+					foreach  ( $categories as $parent ) {
+						$children[] = get_categories(array('child_of'=> $parent));
```
Comment (miguelpeixe) [inline includes/class-newspack-blocks.php:672]:
```suggestion
						$children[] = get_categories( array( 'child_of' => $parent ) );
```

**Review by miguelpeixe — CHANGES_REQUESTED** [review]

File: includes/class-newspack-blocks.php (hunk 1 above)
Comment (DennisNDean) [reply includes/class-newspack-blocks.php]: Ah yes, that's fixed, thank you

File: includes/class-newspack-blocks.php [hunk 4]
```diff
@@ -666,6 +667,14 @@ function ( $acc, $block ) use ( $block_name ) {
 				);
 			}
 			if ( $categories && count( $categories ) ) {
+				if ( $include_subcategories === 1 ){
+					foreach  ( $categories as $parent ) {
```
Comment (miguelpeixe) [inline includes/class-newspack-blocks.php:671]:
```suggestion
					foreach ( $categories as $parent ) {
```

**Review by miguelpeixe — APPROVED** [review]

**Comment by miguelpeixe** [comment]
Thank you for the revisions, @DennisNDean!

**Pull request merged**